UPGRADING.md

```diff
@@ -0,0 +1,56 @@
+This document provides instructions for upgrading between major
+versions of `@lifeomic/one-query`
+
+### 3.x -> 4.x
+
+- Replace your installation of `@tanstack/react-query@4.x` with `@tanstack/react-query@5.x`
+
+- See `react-query`'s [migration guide](https://tanstack.com/query/v5/docs/framework/react/guides/migrating-to-v5) for an idea. The largest changes will be outlined below - however since `one-query` is mostly a typed wrapper around `react-query` most, if not all, things mentioned apply
+
+- The `loading` status has been renamed to `pending`, and similarly the derived `isLoading` flag has been renamed to `isPending`
+
+```diff
+- query.status === 'loading'
++ query.status === 'pending'
+
+- if (query.isLoading)
++ if (query.isPending)
+```
+
+- `useInfiniteAPIQuery` now requires `initialPageParam` and `getNextPageParam` options. Passing page param through `fetchNextPage` and `fetchPreviousPage` is no longer supported.
+
+```diff
+-  const query = useInfiniteAPIQuery('GET /list', {
+-   filter: 'some-filter',
+-   after: undefined,
+- });
+- await void query.fetchNextPage({ pageParam: {...} });
++ const query = useInfiniteAPIQuery(
++   'GET /list',
++   {
++     filter: 'some-filter',
++     after: undefined,
++   },
++   {
++     initialPageParam: {},
++     getNextPageParam: (lastPage) => ({
++       after: lastPage.next,
++     }),
++   },
++ );
++ await void query.fetchNextPage();
+```
+
+- The minimum required TypeScript version is now 4.7
+
+- TypeScript: `Error` is now the default type for errors instead of `unknown`
```

Comment:
that's cool

Reply:
for sure! we could even globally override the default error value, if someone wanted to keep `unknown`, or to an `AxiosError`

https://tanstack.com/query/v5/docs/framework/react/typescript#registering-a-global-error

since do rely on axios for this package we could make that assumption I suppose

Reply:
Cool!